cloud/azure/main.tf

```diff
@@ -182,10 +181,43 @@ resource "azurerm_postgresql_database" "civiform" {
   collation           = "English_United States.1252"
 }
 
-resource "azurerm_postgresql_virtual_network_rule" "civiform" {
-  name                                 = "sqlvnetrule"
-  resource_group_name                  = azurerm_resource_group.rg.name
-  server_name                          = azurerm_postgresql_server.civiform.name
-  subnet_id                            = azurerm_subnet.server_subnet.id
-  ignore_missing_vnet_service_endpoint = false
-}
\ No newline at end of file
+# Configure private link
+resource "azurerm_subnet" "postgres_subnet" {
+  name                 = "postgres_subnet"
+  resource_group_name  = azurerm_resource_group.rg.name
+  virtual_network_name = azurerm_virtual_network.civiform_vnet.name
+  address_prefixes     = ["10.0.4.0/24"]
+
+  enforce_private_link_endpoint_network_policies = true
+}
+
+resource "azurerm_private_dns_zone" "privatelink" {
+  name                = "privatelink.postgres.database.azure.com"
+  resource_group_name = azurerm_resource_group.rg.name
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "vnet_link" {
+  name                  = "vnet_link"
```

Comment:
suggestion (non-blocking): would be good to have a consistent naming convention for the Azure-side names of things. Maybe changing this one to match the convention used on L195?

Reply:
We can try to standardize these as much as possible, but each resource has its own naming rules. Database servers and App Service webapps for example only accept alphanumeric characters and hyphens. Storage accounts only accept alphanumeric characters. On L195, the name had to be a valid domain name: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/private_dns_zone

Reply:
😭

thanks for looking into this